tests/functional/simple_seed/test_seed.py

```diff
@@ -1,19 +1,290 @@
+            },
+        }
+
+    def test_postgres_simple_seed_with_disabled(self, project):
+        results = run_dbt(["seed"])
+        len(results) == 2
+        check_table_does_exist(project.adapter, "seed_enabled")
+        check_table_does_not_exist(project.adapter, "seed_disabled")
+        check_table_does_exist(project.adapter, "seed_tricky")
+        # TODO: 1/3 is there a better way to enforce cleanup
+        project.run_sql(f"drop table {project.test_schema}.seed_enabled")
+        project.run_sql(f"drop table {project.test_schema}.seed_tricky")
+
+    def test_postgres_simple_seed_selection(self, project):
+        results = run_dbt(["seed", "--select", "seed_enabled"])
+        len(results) == 1
+        check_table_does_exist(project.adapter, "seed_enabled")
+        check_table_does_not_exist(project.adapter, "seed_disabled")
+        check_table_does_not_exist(project.adapter, "seed_tricky")
+        # TODO: 2/3
+        project.run_sql(f"drop table {project.test_schema}.seed_enabled")
+
+    def test_postgres_simple_seed_exclude(self, project):
+        results = run_dbt(["seed", "--exclude", "seed_enabled"])
+        len(results) == 1
+        check_table_does_not_exist(project.adapter, "seed_enabled")
+        check_table_does_not_exist(project.adapter, "seed_disabled")
+        check_table_does_exist(project.adapter, "seed_tricky")
+        # TODO: 3/3
+        project.run_sql(f"drop table {project.test_schema}.seed_tricky")
+
+
+class TestSeedParsing(SeedConfigBase):
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self, project):
+        """Create table for ensuring seeds and models used in tests build correctly"""
+        project.run_sql_file(project.test_data_dir / Path("seed_expected.sql"))
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"seed.csv": seeds__wont_parse}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__from_basic_seed}
+
+    def test_postgres_dbt_run_skips_seeds(self, project):
+        # run does not try to parse the seed files
+        len(run_dbt()) == 1
+
+        # make sure 'dbt seed' fails, otherwise our test is invalid!
+        run_dbt(["seed"], expect_pass=False)
+
+
+class TestSimpleSeedWithBOM(SeedConfigBase):
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self, project):
+        """Create table for ensuring seeds and models used in tests build correctly"""
+        project.run_sql_file(project.test_data_dir / Path("seed_expected.sql"))
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        seed_bom = read_file(dir_path, "data", "seed_bom.csv")
+        return {"seed_bom.csv": seed_bom}
+
+    def test_postgres_simple_seed(self, project):
+        # first make sure nobody "fixed" the file by accident
+        seed_path = project.test_data_dir / Path("seed_bom.csv")
```

Comment:
Did the 'read_file' utility not work?

Reply:
It was my thought you had to keep it as is because you need a file object and not a str (which is what read_file returns, when I tested the utility you suggest here).